### PR TITLE
Fix missing atlas initialization and update dependencies

### PR DESCRIPTION
**Pull request by tehrengruber**
Based on some of the changes by @wdeconinck [here](https://github.com/wdeconinck/atlas4py/tree/feature/pointcloud).

### CODE REVIEW

File: .github/workflows/tox.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
         pyversion_no_dot="${{ matrix.python-version }}"
         pyversion_no_dot="${pyversion_no_dot/./}"
         pip install tox
-        tox -v -r -e py${pyversion_no_dot} .
+        tox run -v -e py${pyversion_no_dot} -- .
```
Comment (egparedes) [inline .github/workflows/tox.yml:30]: To make sure you always get the expected behavior using a modern `tox` version, I would recommend to bump `tox` requirement to `tox>=4.0`, both in `requirements-dev.txt` and the `min_version` setting in `tox.ini` (see: https://tox.wiki/en/4.11.4/config.html#min_version)

File: setup.py [hunk 2]
```diff
@@ -120,8 +120,7 @@ def build_extension(self, ext):
         build_args = ["--config", cfg, "-j", str(BUILD_JOBS)]
         subprocess.check_call(["cmake", "--build", "."] + build_args, cwd=self.build_temp)
 
-
-PACKAGE_VERSION = "0.26.0.dev14"
+PACKAGE_VERSION = "0.35.1.dev14"
```
Comment (egparedes) [inline setup.py:123]: Since you update the base atlas version, I would reset the `dev` part to `0` (or `1` depending on your preference).

**Review by egparedes — COMMENTED** [review]
It looks good. I have a couple of suggestions to avoid problems in corner cases, but otherwise is ready to be merged.

**Review by wdeconinck — APPROVED** [review]
Thanks! I like your atexit trick!

**Comment by wdeconinck** [comment]
By the way, I realise I still need to create a PR with my extended bindings. This is my first objective after New Year.

**Review by egparedes — APPROVED** [review]
ILGTM, thanks!

**Pull request merged**